contracts/modules/Checkpoint/EtherDividendCheckpoint.sol

```diff
@@ -0,0 +1,190 @@
+pragma solidity ^0.4.23;
+
+import "./ICheckpoint.sol";
+import "../../interfaces/ISecurityToken.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/math/Math.sol";
+
+contract EtherDividendCheckpoint is ICheckpoint {
+    using SafeMath for uint256;
+
+    struct Dividend {
+      uint256 checkpointId;
+      uint256 created; // Time at which the dividend was created
+      uint256 maturity; // Time after which dividend can be claimed - set to 0 to bypass
+      uint256 expiry;  // Time until which dividend can be claimed - after this time any remaining amount can be withdrawn by issuer - set to very high value to bypass
+      uint256 amount; // Dividend amount in WEI
+      uint256 claimedAmount; // Amount of dividend claimed so far
+      uint256 totalSupply; // Total supply at the associated checkpoint (avoids recalculating this)
+      bool reclaimed;
+      mapping (address => bool) claimed; // List of addresses which have claimed dividend
+    }
+
+    // List of all dividends
+    Dividend[] public dividends;
+
+    event EtherDividendDeposited(address indexed _depositor, uint256 _checkpointId, uint256 _created, uint256 _maturity, uint256 _expiry, uint256 _amount, uint256 _totalSupply, uint256 _dividendIndex);
+    event EtherDividendClaimed(address indexed _payee, uint256 _dividendIndex, uint256 _amount);
+    event EtherDividendReclaimed(address indexed _claimer, uint256 _dividendIndex, uint256 _claimedAmount);
+
+    modifier validDividendIndex(uint256 _dividendIndex) {
+        require(_dividendIndex < dividends.length, "Incorrect dividend index");
+        require(now >= dividends[_dividendIndex].maturity, "Dividend maturity is in the future");
+        require(now < dividends[_dividendIndex].expiry, "Dividend expiry is in the past");
+        require(dividends[_dividendIndex].reclaimed == false, "Dividend has been reclaimed by issuer");
+        _;
+    }
+
+    /**
+     * @dev Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress) public
+    IModule(_securityToken, _polyAddress)
+    {
+    }
+
+    /**
+    * @dev Init function i.e generalise function to maintain the structure of the module contract
+    * @return bytes4
+    */
+    function getInitFunction() public returns(bytes4) {
+        return bytes4(0);
+    }
+
+    /**
+     * @dev Creates a dividend and checkpoint for the dividend
+     * @param _maturity Time from which dividend can be paid
+     * @param _expiry Time until dividend can no longer be paid, and can be reclaimed by issuer
+     */
+    function createDividend(uint256 _maturity, uint256 _expiry) payable public onlyOwner {
+        require(_expiry > _maturity);
```

Comment:
require (msg.value > 0) to avoid "empty" dividend checkpoints

Reply:
Agreed - done.

Reply:
Also added:
`require(_amount > 0);`
on the erc20 side.